crates/derive/src/combine_api.rs

```diff
@@ -177,9 +194,27 @@ impl cgo::Service for API {
                     arena,
                     out,
                     &mut state.stats.out,
+                    &mut state.scratch_shape,
                 )?;
 
                 if !more {
+                    if state.shape.ne(&state.scratch_shape) {
+                        // Update the "true" shape with the newly widened shape
+                        state.shape = state.scratch_shape.clone();
+
+                        match serde_json::to_value(
+                            &SchemaBuilder::new(state.scratch_shape.clone()).root_schema(),
+                        ) {
+                            Ok(value) => tracing::info!(
+                                inferred_schema = ?DebugJson(value),
```

Comment:
this needs to include the collection, for the derivation to then be able to re-map it. That's going to require adding it to the protobuf configuration message.

nit: `schema` as the field name, instead of `inferred_schema`?

Reply:
Plumbed through, and renamed to `schema` 👍